### ISSUE TITLE
Natives are tied to the Browsercraft project

### ISSUE THREAD

**Issue by skyfallwastaken** [leaningtech/cheerpj-natives#2]
https://github.com/leaningtech/cheerpj-natives/blob/main/natives/lwjgl.js#L4

I can see a couple solutions:
- Require loading in the module first, calling a function to set the element, and taking advantage of the fact that the module won't be reloaded when CJ also imports it (probably the best way)
- Doing something like this:
```ts
// set before running the app in CJ
window.lwjglCanvasElem = document.querySelector("minecraft-client").shadowRoot.querySelector("canvas");

// in lwjgl.js
var glCanvas = window.lwjglCanvasElem
if (!glCanvas) {
  throw new Error("GL canvas is not set - have you set `window.lwjglCanvasElem` to your canvas element?")
}
```

**Comment by bates64** [comment]
Definitely prefer (1).

**Comment by skyfallwastaken** [comment]
@bates64 sorry for the ping, but is there any progress on this?

**Comment by bates64** [comment]
No problem! I'd implement this myself but there is a lot on my plate right now.  I'd be happy to take a PR implementing your first suggestion, and I can upgrade browsercraft. It's a little awkward of a solution but I don't see a better way to solve it.

**Comment by bates64** [comment]
Turns out (1) as implemented in #6 doesn't work, because CheerpJ loads the module via a blob URL since the `lwjgl.js` comes from its internal filesystem rather than directly from the `lwjgl.js` URL. Because of this indirection, the JS engine considers the module imported from the blob URL and the module imported from the actual URL to be distinct modules, so we can't share state between them.
